Modules/Filtering/Path/include/itkContourExtractor2DImageFilter.hxx

```diff
@@ -486,7 +486,8 @@ ContourExtractor2DImageFilter<TInputImage>::AddSegment(VertexType from, VertexTy
         // Now remove the old end of 'head' from the ends map and add
         // the new end.
         contourData.m_ContourEnds.erase(newHead);
-        contourData.m_ContourEnds.insert(VertexToContourContainerIteratorMapKeyValuePair(head->back(), head));
+        const VertexToContourContainerIteratorMapKeyValuePair newEnd(head->back(), head);
+        contourData.m_ContourEnds.insert(newEnd);
```

Comment:
Hello, Thank you for addressing the issue about the valgrind defects I created. 

I'm sure you PR is just fine the way it is... but I thought I'd give one of those extra suggestions :)
Consider:

`contourData.m_ContourEnds.emplace(head->back(), head)`

Reply:
Good, thank you.  I have now done that.